### PR TITLE
Config file parsing & saving

### PR DESCRIPTION
**Pull request by MrAksel**

### CODE REVIEW

**Comment by MrAksel** [comment]
Known issue: It reads the config file fine no matter the order of the options, but writes them in a fixed order. That both removes comments and custom formatting, which is bad, but it would be much harder to implement correctly.

**Comment by charlesmilette** [comment]
We could fix the issue with INI, but then we all know how that turned out...

**Comment by MrAksel** [comment]
@olliethepikachu If the user specifies --config "Downloads\myfunnyconfig.cfg" that config should be loaded first, and then the rest of the command line options should be parsed and override the config file. Optionally we could just disable other cmd-options if --config is specified. Just demand that everything is decided there if they provide a custom file.

**Comment by olliethepikachu** [comment]
What if you read from a default config? (If it works like that).

**Comment by ethanhs** [comment]
I think command line switches should supersede settings files (allows for experimentation).

The best way to read the config is loop while there are lines left, then do an if/else for each option. That way the lines can be in any order.

Also, I think it would be nice to have a default local load path. So if there exists a `config.cfg` in the same directory, load that, even if config isn't passed in the command line

**Comment by MrAksel** [comment]
That's how it works now. It loads config.cfg in the current working directory if none are specified, and loops through line by line and has an if/else if to parse the arguments.

File: TranslucentTB/config.cfg [hunk 1]
```diff
@@ -0,0 +1,3 @@
+accent=transparent
```
Comment (ethanhs) [inline TranslucentTB/config.cfg]: I would prefer the default config to be blur, as this has been the default state.

File: TranslucentTB/main.cpp [hunk 2]
```diff
@@ -153,67 +169,185 @@ void PrintHelp()
...
+	{
+		opt.color = (forcedtransparency << 24) +
+			(opt.color & 0x00FFFFFF);
+	}
+}
+
+void SaveConfigFile()
+{
+	using namespace std;
+	if (!configfile.empty())
```
Comment (ethanhs) [inline TranslucentTB/main.cpp]: I think we should only save config is it is passed via command line (I will add a tray option too). Let the user tell it when to overwrite!

**Comment by ethanhs** [comment]
Left a couple of comments. Looks pretty good overall.

**Comment by MrAksel** [comment]
Should have answered to those comments in that commit. I'm planning one more commit to solve cmd-line/config file conflicts. The --config option is handled first, loading the specified config file, and then the rest of the arguments are parsed.

**Comment by MrAksel** [comment]
tl;dr: Only saves to the config file if the user changed mode in the tray. Supports reading several config files, and the last one is used as the save location in case of save needed. The config files are read before the command line is processed, so the command line options will override the config file options.

**Comment by ethanhs** [comment]
LGTM, thanks!
